nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs

```diff
@@ -145,8 +197,16 @@ private void StartWatcher()
         /// <summary>
         /// Gets the list of serial ports.
         /// </summary>
-        /// <returns>The list of serial ports.</returns>
         public List<string> GetPortNames()
+        {
+            return DoGetPortNames();
+        }
+
+        /// <summary>
+        /// Gets the list of serial ports.
+        /// </summary>
+        /// <returns>The list of serial ports that may be connected to a nanoDevice.</returns>
+        internal static List<string> DoGetPortNames()

```

Comment:
I don't think this is needed... Why not keep the GetPortNames() and up return filter out the ports to exclude?
We don't need a cluttered API, rather a simple one. 😉

Reply:
The reason to do this:

- All external code has to now about Port*Manager and should not interact with the DeviceWatchers. If I had designed this library, I would have made the DeviceWatchers internal as they do not work properly unless managed by the Port*Manager.

- External code (e.g., the test framework) that wants to get a list of available ports should use the same logic as the debugger, to exclude all special ports. But that should be a call to a static method, not `new DeviceWatcher(portmanager).GetPortNames ()`.

- So I added a public static method to PortSerialManager and the internal DeviceWatcher.DoGetPortNames is used by both the DeviceWatcher and the static method of PortSerialManager.

Reply:
I get that. 
My point being why do we need both: `public List<string> GetPortNames()` and `internal static List<string> DoGetPortNames()`? `GetPortNames()` is not being called anywhere and currently is just calling `DoGetPortNames()`.
Why not just have `public static List<string> GetPortNames()`, renaming it from `DoGetPortNames` and get rid of the current `GetPortNames()`?

Reply:
Aha. In general I don't change public functions in libraries, because you'll neven know who is using it. In this case that may be a bit too cautious. No other nanoFramework code is calling this method, and I don't think there are many people with custom tools based on this library. I'll make GetPortNames static and remove it from the PortSerialManager.

Reply:
That's a good approach of course, in this case, as you've mentioned it, it should be pretty safe to do that.
Anyway we can always mark a method as obsolete and remove it in a future version. We've done that a number of times.
Feel free to discuss any alternatives that you find viable!